File: 08-number-guesser/num_guess.exs
# Number guessing game written in Elixir
# Created by M.V.Harish Kumar on 14-01-2024

defmodule NumGuess do
    def game() do
        game(:rand.uniform(100), 1, [])
    end

    def game(rnum, guesses, guessed_nums) do
        {unum, _} = IO.gets("Enter a number: ") |> Integer.parse

        cond do
            unum == rnum ->
                IO.puts "Yay! You've guessed the number in #{guesses} chances"
            
            (unum > 100) or (unum < 1) ->
                IO.puts "Guessed number is out of range"
                game(rnum, guesses+1, [unum|guessed_nums])
            
            true ->
                if unum < rnum do
                    IO.puts "You've guessed low"
                else
                    IO.puts "You've guessed high"
                end

                incr = Enum.member?(guessed_nums, unum) && 0 || 1
                game(rnum, guesses+incr, [unum|guessed_nums])
        end
    end
end

IO.puts "Number Guessing Game"
IO.puts "Guess the number which the computer thinks of"
IO.puts "The number is between 1 to 100"
NumGuess.game()

File: 08-number-guesser/num_guess.rb
# Number guessing game in Ruby
# Written by M.V.Harish Kumar on 26-01-2025

rnum = rand(100) + 1
guesses = 0
guessed_nums = []

  puts "Number Guessing Game"
  puts "Guess the number which the computer thinks of"
  puts "The number is between 1 to 100"
  
loop do
  print "Enter your guess: "
  unum = gets.to_i

  puts "Guessed number is out of range" if (unum > 100 || unum < 1)
  case unum <=> rnum
    when -1
      puts "You've guessed low"
      guesses += guessed_nums.include?(unum) ? 0 : 1
    when 0
      guesses += 1
      puts "Yay! you've guessed the number in #{guesses} guesses"
      break
    when 1
      puts "You've guessed high"
      guesses += guessed_nums.include?(unum) ? 0 : 1
  end
  guessed_nums << unum if !guessed_nums.include?(unum)
end
